src/platforms/hosted/jlink.c

```diff
@@ -326,6 +352,9 @@ static bool jlink_get_capabilities(void)
 	jlink.capabilities = read_le4(buffer, 0);
 	DEBUG_INFO("Capabilities: 0x%08" PRIx32 "\n", jlink.capabilities);
 
+	if (jlink.capabilities & JLINK_CAPABILITY_EXTENDED_CAPABILITIES)
+		jlink_get_extended_capabilities();
```

Comment:
Instead of printing the capabilities here, and then duplicating the information printing the extended capabilities, move this print to after the extended capabilities check, and do an early return there

Also, don't forget to return the result, since we are delegating to `jlink_get_extended_capabilities()` we should propagate the result

```suggestion
	if (jlink.capabilities & JLINK_CAPABILITY_EXTENDED_CAPABILITIES)
		return jlink_get_extended_capabilities();
	
	DEBUG_INFO("Capabilities: 0x%08" PRIx32 "\n", jlink.capabilities);
```

Reply:
Agreed. I wrote it that way to see both on BMDA verbose stdout. Since extended capabilities is strictly a superset of capabilities, then it being printed should suppress printing ordinary ones. I understand that this code needs to handle both firmware types.